### PR TITLE
Qradar

### PR DESCRIPTION
**Pull request by barchen1**
From version 5.5.0
## Status
In progress

## Related Issues
fixes: https://github.com/demisto/etc/issues/21207
https://github.com/demisto/etc/issues/21402

## Description
Add upload indicators command that add indicators from Demisto to QRadar reference set according to the user query
Also, I added to the command update reference set value an option to add multiple values and not just one

## Does it break backward compatibility?
     No

## Must have
- [x] Tests
- [ ] Documentation 

## Technical Writer
- [ ] [YAML](https://github.com/demisto/content/edit/Qradar/Integrations/QRadar/QRadar.yml?pr=%2Fdemisto%2Fcontent%2Fpull%2F5519)
- [ ] [CHANGELOG](https://github.com/demisto/content/edit/Qradar/Integrations/QRadar/CHANGELOG.md?pr=%2Fdemisto%2Fcontent%2Fpull%2F5519)

### CODE REVIEW

File: Integrations/QRadar_5_4_9/QRadar_test_5_4_9.py [hunk 1]
```diff
@@ -0,0 +1,467 @@
+import pytest
```
Comment (ronykoz) [inline Integrations/QRadar_5_4_9/QRadar_test_5_4_9.py]: i think that the file itself should not be with the 5_4_9

File: Integrations/QRadar_5_4_9/QRadar_5_4_9.py [hunk 2]
```diff
@@ -0,0 +1,1095 @@
...
+USERNAME = CREDENTIALS['identifier'] if CREDENTIALS else ''
+PASSWORD = CREDENTIALS['password'] if CREDENTIALS else ''
+TOKEN = demisto.params().get('token')
+USE_SSL = not demisto.params().get('insecure', False)
+AUTH_HEADERS = {'Content-Type': 'application/json'}
+if TOKEN:
+    AUTH_HEADERS['SEC'] = str(TOKEN)
+OFFENSES_PER_CALL = int(demisto.params().get('offensesPerCall', 50))
+OFFENSES_PER_CALL = 50 if OFFENSES_PER_CALL > 50 else OFFENSES_PER_CALL
+PAGE_SIZE = 200
```
Comment (ronykoz) [inline Integrations/QRadar_5_4_9/QRadar_5_4_9.py]: isn;t it something you've added in the new version? This file should look like the one from Master

File: Integrations/QRadar_5_4_9/CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+## [Unreleased]
+ - Added **Full Incident Enrichment** instance parameter. Clear this checkbox to disable QRadar offense enrichment performed in fetch-incidents. This might help if you encounter a timeout while fetching new incidents.
```
Comment (ronykoz) [inline Integrations/QRadar_5_4_9/CHANGELOG.md]: are you sure this supposed to be here?

File: Integrations/QRadar/QRadar.yml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 category: Analytics & SIEM
 commonfields:
-  id: QRadar
+  id: IBM QRadar 5.5
```
Comment (ronykoz) [inline Integrations/QRadar/QRadar.yml]: Don't change the ID

File: Integrations/QRadar/QRadar.yml [hunk 5]
```diff
@@ -40,15 +40,15 @@ configuration:
...
+  display: Full Incident Enrichment
   hidden: false
   name: full_enrich
   required: false
   type: 8
 description: Fetch offenses as incidents and search QRadar
-display: IBM QRadar
-name: QRadar
+display: IBM QRadar 5.5
+name: IBM QRadar 5.5
```
Comment (ronykoz) [inline Integrations/QRadar/QRadar.yml]: same here

File: Integrations/QRadar/QRadar.py [hunk 6]
```diff
@@ -974,12 +976,20 @@ def delete_reference_set_command():
...
+        The function creates or updates values in QRadar reference set
+    """
     args = demisto.args()
+    values = argToList(args.get('value'))
     if args.get('date_value') == 'True':
-        value = date_to_timestamp(args.get('value'), date_format="%Y-%m-%dT%H:%M:%S.%f000Z")
+        values = [date_to_timestamp(value, date_format="%Y-%m-%dT%H:%M:%S.%f000Z") for value in values]
+    if len(values) > 1:
+        raw_ref = upload_indicators_list_request(args.get('ref_name'), values)
+        demisto.results(raw_ref)
```
Comment (ronykoz) [inline Integrations/QRadar/QRadar.py]: remove

File: Integrations/QRadar/QRadar.py [hunk 7]
```diff
@@ -974,12 +976,20 @@ def delete_reference_set_command():
...
+    """
+        The function creates or updates values in QRadar reference set
+    """
     args = demisto.args()
+    values = argToList(args.get('value'))
     if args.get('date_value') == 'True':
-        value = date_to_timestamp(args.get('value'), date_format="%Y-%m-%dT%H:%M:%S.%f000Z")
+        values = [date_to_timestamp(value, date_format="%Y-%m-%dT%H:%M:%S.%f000Z") for value in values]
+    if len(values) > 1:
+        raw_ref = upload_indicators_list_request(args.get('ref_name'), values)
```
Comment (ronykoz) [inline Integrations/QRadar/QRadar.py]: why are we calling different functions here and in the elf

File: Integrations/QRadar/QRadar.py [hunk 8]
```diff
@@ -1038,6 +1048,116 @@ def get_domains_by_id_command():
...
+    page = 0
+    while last_found_len == PAGE_SIZE:
+        fetched_iocs = demisto.searchIndicators(query=indicator_query, page=page, size=PAGE_SIZE).get('iocs')
+        for indicator in fetched_iocs:
+            value_indicators_list.append(indicator['value'])
+            indicators_data_list.append({
+                'value': indicator['value'],
+                'indicator_type': indicator['indicator_type']
+            })
+        last_found_len = len(fetched_iocs)
```
Comment (ronykoz) [inline Integrations/QRadar/QRadar.py]: you don't need this

File: Integrations/QRadar/QRadar.py [hunk 9]
```diff
@@ -1038,6 +1048,116 @@ def get_domains_by_id_command():
...
+    last_found_len = PAGE_SIZE
+    value_indicators_list = []
+    indicators_data_list = []
+    page = 0
+    while last_found_len == PAGE_SIZE:
+        fetched_iocs = demisto.searchIndicators(query=indicator_query, page=page, size=PAGE_SIZE).get('iocs')
+        for indicator in fetched_iocs:
+            value_indicators_list.append(indicator['value'])
+            indicators_data_list.append({
+                'value': indicator['value'],
```
Comment (ronykoz) [inline Integrations/QRadar/QRadar.py]: what does the value contain?

File: Integrations/QRadar/QRadar.py [hunk 10]
```diff
@@ -1038,6 +1048,116 @@ def get_domains_by_id_command():
...
+
+    Args:
+        ref_set_name (str): Reference set name
+
+    Returns:
+        Dict. If found - Reference set object, else - Error
+    """
+
+    try:
+        return get_reference_by_name(ref_set_name)
```
Comment (ronykoz) [inline Integrations/QRadar/QRadar.py]:
```suggestion
        return True if get_reference_by_name(ref_set_name) else False
```
Or change the name of the function

File: Integrations/QRadar/QRadar.py [hunk 11]
```diff
@@ -1038,6 +1048,116 @@ def get_domains_by_id_command():
...
+    except Exception as e:
+        if '1002' in str(e):
```
Comment (ronykoz) [inline Integrations/QRadar/QRadar.py]: ? add an explanation

File: Integrations/QRadar/QRadar.py [hunk 12]
```diff
@@ -1038,6 +1048,116 @@ def get_domains_by_id_command():
...
+    """
+        The function finds indicators according to user query and updates QRadar reference set
+    """
+    try:
+        args = demisto.args()
+        reference_name = args.get('ref_name')
+        element_type = args.get('element_type')
+        if not check_ref_set_exist(reference_name):
+            if element_type:
+                create_reference_set(reference_name, args.get('element_type'), args.get('timeout_type'),
```
Comment (ronykoz) [inline Integrations/QRadar/QRadar.py]: Not sure that this a good behaviour, do we have another command to create ref set? if so just error out and point the user to use it, otherwise a simple typo will cause creating a new one

File: Integrations/QRadar/QRadar.py [hunk 13]
```diff
@@ -1038,6 +1048,116 @@ def get_domains_by_id_command():
...
+                             " Try again".format(reference_name))
+        query = args.get('query')
+        value_indicators_list, indicators_data_list = get_indicators_list(query)
+        if len(value_indicators_list) == 0:
+            return {
+                'Type': entryTypes['note'],
+                'Contents': {},
+                'ContentsFormat': formats['json'],
+                'ReadableContentsFormat': formats['markdown'],
+                'HumanReadable': "No indicators found, Reference set {0} didn't change".format(reference_name)
```
Comment (ronykoz) [inline Integrations/QRadar/QRadar.py]: use return_outputs, here and everywhere else

File: Integrations/QRadar/QRadar.py [hunk 14]
```diff
@@ -1038,6 +1048,116 @@ def get_domains_by_id_command():
...
+                'Contents': raw_response,
+                'ContentsFormat': formats['json'],
+                'ReadableContentsFormat': formats['markdown'],
+                'HumanReadable':
+                    tableToMarkdown("reference set {0} was updated".format(reference_name), ref_set_data,
+                                    headers=ref_set_headers) + tableToMarkdown("Indicators list",
+                                                                               indicators_data_list,
+                                                                               headers=indicator_headers)}
+    except Exception as e:
+        if '1005' in str(e):
```
Comment (ronykoz) [inline Integrations/QRadar/QRadar.py]: ? add a comment

File: Integrations/QRadar/QRadar.py [hunk 15]
```diff
@@ -1038,6 +1048,116 @@ def get_domains_by_id_command():
...
+              indicator_query (str): The query demisto.searchIndicators use to find indicators
+        Returns:
+             list of indicators value and a list with all indicators data
+    """
+    last_found_len = PAGE_SIZE
+    value_indicators_list = []
+    indicators_data_list = []
+    page = 0
+    while last_found_len == PAGE_SIZE:
+        fetched_iocs = demisto.searchIndicators(query=indicator_query, page=page, size=PAGE_SIZE).get('iocs')
```
Comment (ronykoz) [inline Integrations/QRadar/QRadar.py]: we need pagination here, otherwise the user will get stuck with the limit he entered

File: Integrations/QRadar/QRadar.yml (hunk 4 above)
Comment (barchen1) [reply Integrations/QRadar/QRadar.yml]: the original ID was IBM QRadar,
Do you want me to remove 5.5 ?

File: Integrations/QRadar/QRadar.yml (hunk 5 above)
Comment (barchen1) [reply Integrations/QRadar/QRadar.yml]: the original ID was IBM QRadar,
Do you want me to remove 5.5 ?

File: Integrations/QRadar_5_4_9/CHANGELOG.md (hunk 3 above)
Comment (barchen1) [reply Integrations/QRadar_5_4_9/CHANGELOG.md]: Ill change it

File: Integrations/QRadar/QRadar.yml [hunk 16]
```diff
@@ -1287,7 +1287,7 @@ script:
     - default: false
       description: 'The value to add or update in the reference set. Note: Date values
         must be represented in milliseconds since the Unix Epoch January 1st 1970.'
-      isArray: false
+      isArray: true
       name: value
```
Comment (reutshal) [inline Integrations/QRadar/QRadar.yml]:
```suggestion
      name: values
```

File: Integrations/QRadar/QRadar.yml [hunk 17]
```diff
@@ -1234,7 +1234,7 @@ script:
         Epoch January 1st 1970). If ''date_value'' is set to ''True'', then the argument
         will be converted from date in format: ''%Y-%m-%dT%H:%M:%S.%f000Z'' (e.g.
         ''2018-11-06T08:56:41.000000Z'') to epoch.'
-      isArray: false
+      isArray: true
       name: value
```
Comment (reutshal) [inline Integrations/QRadar/QRadar.yml]:
```suggestion
      name: values
```

File: Integrations/QRadar/QRadar.yml [hunk 18]
```diff
@@ -1503,6 +1503,76 @@ script:
...
+      - ALNIC
+      - IP
+      - NUM
+      - PORT
+      - DATE
+      required: false
+      secret: false
+    - auto: PREDEFINED
+      default: false
+      description: The allowed values are "FIRST_SEEN", LAST_SEEN and UNKNOWN. The
```
Comment (reutshal) [inline Integrations/QRadar/QRadar.yml]: image

Please follow the templates for 'List of predetermined options'

File: Tests/conf.json [hunk 19]
```diff
@@ -2089,10 +2089,9 @@
           "instance_names": "ewv2_regular"
         },
         {
-          "integrations": "QRadar",
+          "integrations": "IBM QRadar",
           "playbookID": "QRadar Indicator Hunting Test",
-          "timeout": 1200,
-          "fromversion": "5.0.0"
```
Comment (reutshal) [inline Tests/conf.json]: Why did you remove this?

**Comment by reutshal** [comment]
@barchen1 please add the test playbook for 5.5.0

File: Integrations/QRadar/QRadar.py [hunk 20]
```diff
@@ -25,6 +26,7 @@
     AUTH_HEADERS['SEC'] = str(TOKEN)
 OFFENSES_PER_CALL = int(demisto.params().get('offensesPerCall', 50))
 OFFENSES_PER_CALL = 50 if OFFENSES_PER_CALL > 50 else OFFENSES_PER_CALL
+PAGE_SIZE = 200
```
[the remark this answers is not in the document]
Comment (barchen1) [reply Integrations/QRadar/QRadar.py]: done

File: Integrations/QRadar/QRadar.py [hunk 21]
```diff
@@ -1038,6 +1047,106 @@ def get_domains_by_id_command():
...
+    """
+    url = '{0}/api/reference_data/sets/bulk_load/{1}'.format(SERVER, urllib.quote(reference_name, safe=''))
+    params = {'name': reference_name}
+    return send_request('POST', url, params=params, data=json.dumps(indicators_list))
+
+
+def upload_indicators_command():
+    """
+        The function finds indicators according to user query and updates QRadar reference set
+    """
```
[the remark this answers is not in the document]
Comment (barchen1) [reply Integrations/QRadar/QRadar.py]: done

File: Integrations/QRadar/QRadar.py [hunk 22]
```diff
@@ -1038,6 +1047,106 @@ def get_domains_by_id_command():
...
+        args = demisto.args()
+        reference_name = args.get('ref_name')
+        element_type = args.get('element_type')
+        timeout_type = args.get('timeout_type')
+        time_to_live = args.get('time_to_live')
+        limit = int(args.get('limit'))
+        page = int(args.get('page'))
+        if not check_ref_set_exist(reference_name):
+            if element_type:
+                create_reference_set(reference_name, element_type, args.get('timeout_type'),
```
[the remark this answers is not in the document]
Comment (barchen1) [reply Integrations/QRadar/QRadar.py]: done

File: Integrations/QRadar/QRadar.py [hunk 23]
```diff
@@ -1038,6 +1047,106 @@ def get_domains_by_id_command():
...
+                                     args.get('time_to_live'))
```
[the remark this answers is not in the document]
Comment (barchen1) [reply Integrations/QRadar/QRadar.py]: done

File: Integrations/QRadar/QRadar.py [hunk 24]
```diff
@@ -1038,6 +1047,106 @@ def get_domains_by_id_command():
...
+            else:
+                return_error("There isn't a reference set with the name {0}. To create a reference set,"
```
[the remark this answers is not in the document]
Comment (barchen1) [reply Integrations/QRadar/QRadar.py]: done

File: Integrations/QRadar/QRadar.py [hunk 25]
```diff
@@ -1038,6 +1047,106 @@ def get_domains_by_id_command():
...
+                             " you have to enter an element type".format(reference_name))
```
[the remark this answers is not in the document]
Comment (barchen1) [reply Integrations/QRadar/QRadar.py]: done

File: Integrations/QRadar/QRadar.py [hunk 26]
```diff
@@ -1038,6 +1047,106 @@ def get_domains_by_id_command():
...
+        else:
+            if element_type or time_to_live or timeout_type:
+                return_error("The reference set {0} is already exist. You are not supposed to enter element type, "
+                             "time to live or time to live, Try again".format(reference_name))
```
[the remark this answers is not in the document]
Comment (barchen1) [reply Integrations/QRadar/QRadar.py]: done

File: Integrations/QRadar/QRadar.py [hunk 27]
```diff
@@ -1038,6 +1047,106 @@ def get_domains_by_id_command():
...
+        query = args.get('query')
+        value_indicators_list, indicators_data_list = get_indicators_list(query, limit, page)
```
[the remark this answers is not in the document]
Comment (barchen1) [reply Integrations/QRadar/QRadar.py]: done

File: Integrations/QRadar/QRadar.py [hunk 28]
```diff
@@ -1038,6 +1047,106 @@ def get_domains_by_id_command():
...
+        if len(value_indicators_list) == 0:
```
[the remark this answers is not in the document]
Comment (barchen1) [reply Integrations/QRadar/QRadar.py]: done

File: Integrations/QRadar/QRadar.py [hunk 29]
```diff
@@ -1038,6 +1047,106 @@ def get_domains_by_id_command():
...
+            return ["No indicators found, Reference set {0} didn't change".format(reference_name), {}]
+        else:
+            raw_response = upload_indicators_list_request(reference_name, value_indicators_list)
```
[the remark this answers is not in the document]
Comment (barchen1) [reply Integrations/QRadar/QRadar.py]: done

File: Integrations/QRadar/QRadar.py [hunk 30]
```diff
@@ -1038,6 +1047,106 @@ def get_domains_by_id_command():
...
+def get_indicators_list(indicator_query, limit, page):
+    """
+        Get Demisto indicators list using demisto.searchIndicators
+
+        Args:
+              indicator_query (str): The query demisto.searchIndicators use to find indicators
+              limit (int): The amount of indicators the user want to add to reference set
+              page (int): Page's number the user would like to start from
+        Returns:
+             list of indicators value and a list with all indicators data
```
[the remark this answers is not in the document]
Comment (barchen1) [reply Integrations/QRadar/QRadar.py]: done

File: Integrations/QRadar/QRadar.yml (hunk 16 above)
Comment (barchen1) [reply Integrations/QRadar/QRadar.yml]: done

File: Integrations/QRadar/QRadar.yml (hunk 18 above)
Comment (barchen1) [reply Integrations/QRadar/QRadar.yml]: done

File: Integrations/QRadar/QRadar.py [hunk 31]
```diff
@@ -1066,47 +1064,44 @@ def upload_indicators_list_request(reference_name, indicators_list):
 def upload_indicators_command():
     """
         The function finds indicators according to user query and updates QRadar reference set
+
+        Returns:
+            string, dict: Human readable and the raw response
```
Comment (ronykoz) [inline Integrations/QRadar/QRadar.py]:
```suggestion
            (string, dict). Human readable and the raw response
```

File: Integrations/QRadar/QRadar.py (hunk 10 above)
Comment (ronykoz) [reply Integrations/QRadar/QRadar.py]: so update the function name to get_ref_set

Comment (barchen1) [reply Integrations/QRadar/QRadar.py]: done

File: Tests/conf.json (hunk 19 above)
Comment (barchen1) [reply Tests/conf.json]: changed back

**Comment by barchen1** [comment]
@kirbles19 Can you please go over the following files:
CHANGELOG - https://github.com/demisto/content/blob/Qradar/Integrations/QRadar/CHANGELOG.md

YAML- 
https://github.com/demisto/content/blob/Qradar/Integrations/QRadar/QRadar.yml

thanks

File: Integrations/QRadar/QRadar.py [hunk 32]
```diff
@@ -1050,6 +1058,111 @@ def get_domains_by_id_command():
...
+def get_indicators_list(indicator_query, limit, page):
+    """
+        Get Demisto indicators list using demisto.searchIndicators
+
+        Args:
+              indicator_query (str): The query demisto.searchIndicators use to find indicators
+              limit (int): The amount of indicators the user want to add to reference set
+              page (int): Page's number the user would like to start from
+        Returns:
+             list, list: list of indicators values and a list with all indicators data
```
Comment (reutshal) [inline Integrations/QRadar/QRadar.py]:
```suggestion
             list, list: list of indicators values and a list with all indicators data
```

File: TestPlaybooks/playbook-TestQradar_5_4_9.yml [hunk 33]
```diff
@@ -1,5 +1,5 @@
 id: test_Qradar
-version: -1
+version: 15
```
Comment (ronykoz) [inline TestPlaybooks/playbook-TestQradar_5_4_9.yml]:
```suggestion
version: -1
```

Comment (ronykoz) [inline TestPlaybooks/playbook-TestQradar_5_4_9.yml]: make sure that this playbook is 100% the same as the one in master

File: Integrations/QRadar/QRadar.yml [hunk 34]
```diff
@@ -1234,8 +1234,8 @@ script:
         Epoch January 1st 1970). If ''date_value'' is set to ''True'', then the argument
         will be converted from date in format: ''%Y-%m-%dT%H:%M:%S.%f000Z'' (e.g.
         ''2018-11-06T08:56:41.000000Z'') to epoch.'
-      isArray: false
-      name: value
+      isArray: true
+      name: values
```
Comment (ronykoz) [inline Integrations/QRadar/QRadar.yml]: you can't change this name, it will break BC for customers

File: Integrations/QRadar/QRadar.py [hunk 35]
```diff
@@ -1091,6 +1214,8 @@ def get_domains_by_id_command():
         demisto.results(get_domains_command())
     elif demisto.command() == 'qradar-get-domain-by-id':
         demisto.results(get_domains_by_id_command())
+    elif demisto.command() == 'qradar-upload-indicators':
+        return_outputs(upload_indicators_command())
```
Comment (ronykoz) [inline Integrations/QRadar/QRadar.py]: you are not returning anything to this place, please fix it

File: Integrations/QRadar/QRadar.py [hunk 36]
```diff
@@ -1048,6 +1064,113 @@ def get_domains_by_id_command():
...
+                return_error("There isn't a reference set with the name {0}. To create one,"
+                             " please enter an element type".format(reference_name))
+        else:
+            if element_type or time_to_live or timeout_type:
+                return_error("The reference set {0} is already exist. Element type, time to live or timeout type "
+                             "cannot be modified".format(reference_name))
+        query = args.get('query')
+        indicators_values_list, indicators_data_list = get_indicators_list(query, limit, page)
+        if len(indicators_values_list) == 0:
+            return ["No indicators found, Reference set {0} didn't change".format(reference_name), {}]
```
Comment (ronykoz) [inline Integrations/QRadar/QRadar.py]: why is it a list

File: Integrations/QRadar/QRadar.py [hunk 37]
```diff
@@ -1048,6 +1064,113 @@ def get_domains_by_id_command():
...
+             list, list: List of indicators values and a list with all indicators data
+    """
+    indicators_values_list = []
+    indicators_data_list = []
+    fetched_iocs = demisto.searchIndicators(query=indicator_query, page=page, size=limit).get('iocs')
+    for indicator in fetched_iocs:
+        indicators_values_list.append(indicator['value'])
+        indicators_data_list.append({
+            'value': indicator['value'],
+            'indicator_type': indicator['indicator_type']
```
Comment (ronykoz) [inline Integrations/QRadar/QRadar.py]:
```suggestion
            'type': indicator['indicator_type']
```

File: Integrations/QRadar/QRadar.py [hunk 38]
```diff
@@ -984,12 +993,19 @@ def delete_reference_set_command():
 
 
 def update_reference_set_value_command():
+    """
+        The function creates or updates values in QRadar reference set
+    """
     args = demisto.args()
+    values = argToList(args.get('values'))
```
Comment (ronykoz) [inline Integrations/QRadar/QRadar.py]:
```suggestion
    values = argToList(args.get('value'))
```

File: Integrations/QRadar/QRadar.py [hunk 39]
```diff
@@ -984,12 +993,19 @@ def delete_reference_set_command():
 
 
 def update_reference_set_value_command():
```
Comment (ronykoz) [inline Integrations/QRadar/QRadar.py]: why did we change this at all

File: Integrations/QRadar/QRadar.py (hunk 38 above)
Comment (barchen1) [reply Integrations/QRadar/QRadar.py]: done

File: Integrations/QRadar/QRadar.py (hunk 36 above)
Comment (barchen1) [reply Integrations/QRadar/QRadar.py]: done

File: Integrations/QRadar/QRadar.py (hunk 37 above)
Comment (barchen1) [reply Integrations/QRadar/QRadar.py]: done

File: Integrations/QRadar/QRadar.py (hunk 35 above)
Comment (barchen1) [reply Integrations/QRadar/QRadar.py]: done

File: Integrations/QRadar/QRadar.yml [hunk 40]
```diff
@@ -1235,7 +1235,7 @@ script:
         will be converted from date in format: ''%Y-%m-%dT%H:%M:%S.%f000Z'' (e.g.
         ''2018-11-06T08:56:41.000000Z'') to epoch.'
```
[the remark this answers is not in the document]
Comment (barchen1) [reply Integrations/QRadar/QRadar.yml]: done

File: Integrations/QRadar/QRadar.yml (hunk 34 above)
Comment (barchen1) [reply Integrations/QRadar/QRadar.yml]: done

File: Integrations/QRadar_5_4_9/QRadar_5_4_9.py (hunk 2 above)
Comment (ronykoz) [reply Integrations/QRadar_5_4_9/QRadar_5_4_9.py]: ?

Comment (ronykoz) [reply Integrations/QRadar_5_4_9/QRadar_5_4_9.py]: lets make sure it's a copy paste from master, the yml as well

Comment (barchen1) [reply Integrations/QRadar_5_4_9/QRadar_5_4_9.py]: done

**Comment by kirbles19** [comment]
@barchen1 Done.

**Comment by barchen1** [comment]
Fixed: https://github.com/demisto/etc/issues/21402
@David-BMS

**Comment by ronykoz** [comment]
Merging due fromversion addition in the old file
